pkg/asset/installconfig/azure/client.go

```diff
@@ -20,6 +23,7 @@ type API interface {
 	GetControlPlaneSubnet(ctx context.Context, resourceGroupName, virtualNetwork, subnet string) (*aznetwork.Subnet, error)
 	ListLocations(ctx context.Context) (*[]azsubs.Location, error)
 	GetResourcesProvider(ctx context.Context, resourceProviderNamespace string) (*azres.Provider, error)
+	ValidateResourceSkuAvailability(ctx context.Context, region, diskType string) (*azsku.ResourceSku, error)
```

Comment:
this function feels out of place in the API interface, the rest are getters, while this one is validator.

Reply:
maybe  convert this to GetDiskSkus(ctx, region)
and then check in the validate function using the list.